packages/interactivity/src/directives.js

```diff
@@ -267,15 +267,57 @@ export default () => {
 	// data-wp-watch--[name]
 	directive( 'watch', ( { directives: { watch }, evaluate } ) => {
 		watch.forEach( ( entry ) => {
-			useWatch( () => evaluate( entry ) );
+			useWatch( () => {
+				const start = performance.now();
+				const result = evaluate( entry );
+				performance.measure(
```

Comment:
@gziolo Is there any precedent for a build-time debug flag in Gutenberg? I couldn't fine one. 

My goal is the ability to do a debug build that will include this timing code, while having it absent from regular production builds. I expect these calls add some sort of overhead so want to avoid calling hem or even including them if possible. 

I've done this before with a build-time config variable, essentially if webpack knows a constant is not set (like debug_mode) it optimizes the code out. Open to other ideas for how to achieve the goal.

Reply:
Yes, you can use `SCRIPT_DEBUG`:

https://github.com/WordPress/gutenberg/blob/500aaf66cf2340002f6cf6cbb6bc42dc6bef7dc9/tools/webpack/shared.js#L72-L73

It's supported in both Gutenberg and WordPress core. It will include the code only in the development version which reflects how we use `SCRIPT_DEBUG` on the server.

For `@wordpress/warning` we have a Babel plugin that automatically wraps the code with a check:

https://github.com/WordPress/gutenberg/blob/500aaf66cf2340002f6cf6cbb6bc42dc6bef7dc9/packages/warning/test/babel-plugin.js#L31

You should only care about the syntax that works also for projects that don't define `SCRIPT_DEBUG` replacement in their build configuration.

Reply:
Great, thank you. 

I wrapped the instrumentation calls in the SCRIPT_DEBUG check. The linter complained about the `start` variable being unused or undefined (because it is now wrapped in a conditional). I added ignore comments, if preferable I could define `start` outside the conditional check.